### ISSUE TITLE
Payment and Shipping methods not translated

### ISSUE THREAD

**Issue by henrilindeberg** [hyyan/woo-poly-integration#36]
Payment and Shipping method title and description are not translated. How could I make them appear under String translation?

**Comment by decarvalhoaa** [comment]
I'm working on a fix for this. will share in my fork once it is ready.

**Comment by decarvalhoaa** [comment]
I just finished a fix for Payment gateways and Shipment methods translations using Polylang's string translation table. Right now only BACS and Paypal are done - still need to do Cheque and COD.
And didn't had yet time to test it properly.

The latest version can be downloaded here https://github.com/decarvalhoaa/woo-poly-integration/archive/master.zip

Can you check whether this would work for you?

**Comment by Asasergey** [comment]
Thank you very much. Everything works. The lines (bacs_gateway_description - bacs_gateway_instructions и cheque_gateway_description - cheque_gateway_instructions) must be equally filled. Otherwise the changes are not saved.

**Comment by decarvalhoaa** [comment]
Please note that Cheque and Cash On Delivery is not fully tested/working since I don't use that functionality. It is likelly that the Instructions for Cheque & COD will not be translated, only the Title and Description of these gateways. I will implement it when time allows it.

Regarding the need to fill some text for the changes to be saved (I think you means to be registered in the Polylang's Strings translations table), that is how Polylang works. You need a specific string to register the string to be able "call" the translations afterwards from the database. If you want this strings to be empty, you can still define a dummy text in the admin backend and change/delete the respective translations in the Strings Translations table.

**Comment by Asasergey** [comment]
Thank you

**Comment by decarvalhoaa** [comment]
I added support for COD and Cheque gateways (the email and thank you page) to my developement branch. I added a few action hooks so that other plugins can add extensions for custom/3rd party gateways. 

It is not thoroughly tested. I would welcome support to tested, and I will add it to the master branch and setup a new maintenance release for that.

https://github.com/decarvalhoaa/woo-poly-integration/archive/development.zip

**Comment by decarvalhoaa** [comment]
@hyyan You already have a PR that fixes Shipping. Next will port the Payment GW fix. You can assigned this to me, if you want.

**Comment by Jon007** [comment]
thanks @decarvalhoaa just please note outstanding issues 22 and 23 on your branch payment gateway are not reproduced - I'm happy to test this

**Comment by decarvalhoaa** [comment]
Found the issue and is corrected on the new PR. Just waiting for Hyyan to merge my PRs.

**Comment by hyyan** [comment]
Merged @decarvalhoaa PRs
